serialization/serialization.go

```diff
@@ -0,0 +1,263 @@
+// Package serialization is for working with the mysql::serialization format
+//
+// mysql::serialization is a serialization format introduced with tagged GTIDs
+//
+// https://dev.mysql.com/doc/dev/mysql-server/latest/PageLibsMysqlSerialization.html
+package serialization
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/bits"
+	"strings"
+)
+
+// Message is a mysql::serialization message
+type Message struct {
+	Version uint8 // >= 0
+	Format  Format
+}
+
+func (m *Message) String() (text string) {
+	text += fmt.Sprintf("Message (version: %d)", m.Version)
+	for _, line := range strings.Split(m.Format.String(), "\n") {
+		text += "\n  " + line
+	}
+	return
+}
+
+// GetFieldByName returns a field if the name matches and an error if there is no match
+func (m *Message) GetFieldByName(name string) (Field, error) {
+	for _, f := range m.Format.Fields {
+		if f.Name == name {
+			return f, nil
+		}
+	}
+	return Field{}, fmt.Errorf("field not found: %s", name)
+}
+
+// Format is describing a `message_format`
+type Format struct {
+	Size                  uint64
+	LastNonIgnorableField int
```

Comment:
this is 1 byte as described in mysql source code document URL at the beginning of this file, so the type should be `byte` rather than `int`? also seems above Size field can be up to 9 byte 🤔 not sure if I understand it, will check the whole code file soon